src/nft/components/explore/Carousel.tsx

```diff
@@ -0,0 +1,142 @@
+import { useWindowSize } from 'hooks/useWindowSize'
+import { ChevronLeftIcon } from 'nft/components/icons'
+import { ReactNode, useCallback, useEffect, useRef, useState } from 'react'
+import { a, useSprings } from 'react-spring'
+import styled from 'styled-components/macro'
+
+const CarouselContainer = styled.div`
+  display: flex;
+  max-width: 592px;
+  width: 100%;
+`
+
+const CarouselCardContainer = styled.div`
+  max-width: 512px;
+  position: relative;
+  width: 100%;
+  overflow-x: hidden;
+`
+
+const CarouselCard = styled(a.div)`
+  position: absolute;
+  padding-left: 16px;
+  padding-right: 16px;
+  display: flex;
+  top: 3px;
+  height: 280px;
+  will-change: transform;
+  justify-content: center;
+
+  @media only screen and (max-width: ${({ theme }) => `${theme.breakpoint.md}px`}) {
+    height: 296px;
+  }
+`
+
+const IconContainer = styled.div<{ right?: boolean }>`
+  display: flex;
+  height: 280px;
+  justify-content: center;
+  align-items: center;
+  cursor: pointer;
+  padding: 8px;
+  ${({ right }) => (right ? 'transform: rotate(180deg)' : undefined)};
+  color: ${({ theme }) => theme.textPrimary};
+
+  :hover {
+    opacity: ${({ theme }) => theme.opacity.hover};
+  }
+
+  @media only screen and (max-width: ${({ theme }) => `${theme.breakpoint.md}px`}) {
+    height: 296px;
+  }
+`
+
+interface CarouselProps {
+  children: ReactNode[]
+}
+
+const FIRST_CARD_OFFSET = 0
+const MAX_CARD_WIDTH = 512
+
+export const Carousel = ({ children }: CarouselProps) => {
+  const { width } = useWindowSize()
+  const carouselCardContainerRef = useRef<HTMLDivElement>(null)
+  const [cardWidth, setCardWidth] = useState(MAX_CARD_WIDTH)
+
+  useEffect(() => {
+    if (carouselCardContainerRef.current) {
+      setCardWidth(Math.min(carouselCardContainerRef.current.offsetWidth, MAX_CARD_WIDTH))
+    }
+  }, [width])
+
+  const idx = useCallback((x: number, l = children.length) => (x < 0 ? x + l : x) % l, [children])
```

Comment:
many of these callback can be converted to helper functions in an explore util file to help clean up the component